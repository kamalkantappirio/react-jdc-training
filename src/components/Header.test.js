import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from './Header';

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Header />, div)
  });

  it('should render properly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('a')).to.have.length(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle a click event', () => {
    throw 'TODO';
  });
});