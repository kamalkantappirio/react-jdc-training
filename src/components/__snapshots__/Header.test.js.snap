// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App-header"
>
    <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
    />
    <a
        href="/"
        id="logo"
        onClick={undefined}
    >
        <h2>
            Demo Title 
        </h2>
    </a>
</div>,
  "nodes": Array [
    <div
      className="App-header"
>
      <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
      />
      <a
            href="/"
            id="logo"
            onClick={undefined}
      >
            <h2>
                  Demo Title 
            </h2>
      </a>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <a
                    href="/"
                    id="logo"
                    onClick={undefined}
          >
                    <h2>
                              Demo Title 
                    </h2>
          </a>
</div>,
        "_debugID": 8,
        "_renderedOutput": <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <a
                    href="/"
                    id="logo"
                    onClick={undefined}
          >
                    <h2>
                              Demo Title 
                    </h2>
          </a>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
